src/lightgbm/src/main/scala/LightGBMClassifier.scala

```diff
@@ -84,16 +85,66 @@ class LightGBMClassificationModel(
   set(rawPredictionCol, rawPredictionColName)
   if (thresholdValues.isDefined) set(thresholds, thresholdValues.get)
 
-  override protected def raw2probabilityInPlace(rawPrediction: Vector): Vector = {
-    rawPrediction match {
-      case dv: DenseVector =>
-        dv.values(0) = 1.0 / (1.0 + math.exp(-dv.values(0)))
-        dv.values(1) = 1.0 - dv.values(0)
-        dv
-      case sv: SparseVector =>
-        throw new RuntimeException("Unexpected error in LightGBMClassificationModel:" +
-          " raw2probabilityInPlace encountered SparseVector")
+  /**
+    * Implementation based on ProbabilisticClassifier with slight modifications to
+    * avoid calling raw2probabilityInPlace to defer the probability calculation
+    * to lightgbm native code.
+    *
+    * @param dataset input dataset
+    * @return transformed dataset
+    */
+  override def transform(dataset: Dataset[_]): DataFrame = {
+    transformSchema(dataset.schema, logging = true)
+    if (isDefined(thresholds)) {
+      require($(thresholds).length == numClasses, this.getClass.getSimpleName +
+        ".transform() called with non-matching numClasses and thresholds.length." +
+        s" numClasses=$numClasses, but thresholds has length ${$(thresholds).length}")
+    }
+
+    // Output selected columns only.
+    var outputData = dataset
+    var numColsOutput = 0
+    if ($(rawPredictionCol).nonEmpty) {
+      val predictRawUDF = udf { (features: Any) =>
+        predictRaw(features.asInstanceOf[Vector])
+      }
+      outputData = outputData.withColumn(getRawPredictionCol, predictRawUDF(col(getFeaturesCol)))
+      numColsOutput += 1
+    }
+    if ($(probabilityCol).nonEmpty) {
+      val probabilityUDF = udf { (features: Any) =>
+        predictProbability(features.asInstanceOf[Vector])
+      }
+      val probUDF = probabilityUDF(col($(featuresCol)))
```

Comment:
Use getFeaturesCol instead of $ accessors

Reply:
done!